src/application/app.rs

```diff
@@ -52,6 +52,14 @@ impl App {
         }
     }
 
+    pub fn edit_task(&mut self, idx: Option<usize>) {
+        if let Some(i) = idx {
+            if let Ok(task) = Task::from_str(&self.input) {
+                self.tasks[i] = task;
+            }
+        }
```

Comment:
Just a note, if you have some nested if let statements, you can also try the let else pattern in rust
```suggestion
        let Some(i) = idx else { return };
        if let Ok(task) = Task::from_str(&self.input) {
            self.tasks[i] = task;
        }
```

Reply:
Good advice!